version: '3'
services:
  plex:
    container_name: plex
    image: linuxserver/plex:latest
    restart: unless-stopped
    environment:
      - PUID=${PUID} # default user id, defined in .env
      - PGID=${PGID} # default group id, defined in .env
      - TZ=${TZ} # timezone, defined in .env
      - ADVERTISE_IP=192.168.1.31
    volumes:
      - ./config:/config # plex database
      - ${GDRIVE_DIR}:/gdrive # google drive
    labels:
      - 'traefik.enable=true'
      - 'traefik.http.routers.plex.entrypoints=https'
      - "traefik.http.services.plex.loadbalancer.server.port=32400"
      # - networks=bridge
    ports: # https://support.plex.tv/articles/201543147-what-network-ports-do-i-need-to-allow-through-my-firewall/
      # TODO: Try to remove this port and use plex on 443 port.
      - 32400:32400

networks:
  default:
    external:
      name: global_net
